dht/i_partitioner.cc

```diff
@@ -179,6 +179,20 @@ unsigned shard_of(const schema& s, const token& t) {
     return s.get_sharder().shard_of(t);
 }
 
+static bool token_range_is_not_empty(const dht::token_range& r) {
+    bool is_bounded = r.start() && !r.start()->value().is_minimum() && r.end() && !r.end()->value().is_maximum();
```

Comment:
What if we somehow get is_maximum() in start() or is_minimum() in end()?

Reply:
This is a nonwrapping_interval, so if start() is maximum, then end() must be either maximum() or empty/open.
Ditto for the second case.

Reply:
In other words, `!r.start()->value().is_maximum()` is implied by `r.end() && !r.end()->value().is_maximum()`

Reply:
Right. Please add a comment.

Reply:
Done.